packages/xo-web/src/xo-app/menu/index.js

```diff
@@ -79,7 +79,12 @@ export default class Menu extends Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (!isEqual(Object.keys(prevProps.hosts).sort(), Object.keys(this.props.hosts).sort())) {
+    if (
+      !isEqual(
+        map(prevProps.hosts, host => host.power_state),
```

Comment:
We should also watch the `IDs` because we can't be sure that the got hosts keep on the same order and this code doesn't handle the hosts' re-installation.